app/javascript/crayons/MultiInput/MultiInput.jsx

```diff
@@ -0,0 +1,113 @@
+import { h, Fragment } from 'preact';
+import { useRef, useState } from 'preact/hooks';
+
+const KEYS = {
+  ENTER: 'Enter',
+  COMMA: ',',
+  SPACE: ' ',
+};
+// TODO: think about how this may change based on
+// a different usage. We will most likely want this to be passed in as a prop.
+const ALLOWED_CHARS_REGEX = /([a-zA-Z0-9@.])/;
+
+/**
+ * Component allowing users to add multiple inputs that gets displayed as desctructive pills
+ *
+ * @param {Object} props
+ * @param {string} props.placeholder Input placeholder text
+ */
+
+export const MultiInput = ({ placeholder }) => {
+  const inputRef = useRef(null);
+  const [items, setItems] = useState([]);
+
+  const handleBlur = ({ target: { value } }) => {
+    addItemToList(value);
+    clearSelection();
+  };
+
+  const handleKeyDown = (e) => {
+    switch (e.key) {
+      case KEYS.SPACE:
+      case KEYS.ENTER:
+      case KEYS.COMMA:
+        e.preventDefault();
+        addItemToList(e.target.value);
+        clearSelection();
+        break;
+      default:
+        if (!ALLOWED_CHARS_REGEX.test(e.key)) {
+          e.preventDefault();
+        }
+    }
+  };
+
+  const handleDestructiveClick = (clickedItem) => {
+    const newArr = items.filter((item) => item !== clickedItem);
+    setItems(newArr);
+  };
+
+  const addItemToList = (value) => {
+    // TODO: we will want to do some validation here based on a prop
+    if (value.trim().length > 0) {
+      setItems([...items, value]);
+    }
+  };
+
+  const clearSelection = () => {
+    inputRef.current.value = '';
+  };
+
+  const listItems = items.map((item, index) => (
+    <li
+      key={index}
+      class="c-input--multi__selection-list-item w-max"
+      style="order: 1;"
+    >
+      <div role="group" aria-label="two" class="flex mr-1 mb-1 w-max">
```

Comment:
Are we also going to add the "edit in place" feature that we have on the tag autocomplete selections - so when you click on the text rather than the X button, the input moves to that position and is pre-filled with the item text to begin editing it?

If we don't need it for this feature, then we could remove some code (like this group and the `style="order: 1"`)

Reply:
Just confirmed with Anuj now, and based on [the conversation](https://forem-team.slack.com/archives/C01S4QZUDNC/p1654594483016859) I'm going to leave it in for now, until I tackle the Edit functionality in subsequent PRs.